Assets/Scripts/Player/PlayerInput.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerInput: MonoBehaviour
+{
+    private PlayerMovement playerMovement;
+    public bool inputLocked;
+    public float manaPoints;
+    public int maxMana;
+    public delegate void RotateEventDelegate();
```

Comment:
C# has some built-in delegate type you can use if you don't want to always define your own. In this case, you could use Action [(C# docs)](https://docs.microsoft.com/en-us/dotnet/api/system.action?view=net-5.0)